### PR TITLE
Deploy history report

### PR DESCRIPTION
**Pull request by AliRizvi1**
https://github.com/dimagi/commcare-hq/issues/27839
One feature of the above issue.

##### SUMMARY
Moved the Deploy History Report previously found on the 'System Info' admin page to its own standalone page. Part of a larger process of adding a commit field to the deploy history as explained in the above issue.

##### PRODUCT DESCRIPTION
Users will now be able to access the 'deploy history report' on its own standalone page.

### CODE REVIEW

File: corehq/apps/hqadmin/reports.py [hunk 1]
```diff
@@ -266,3 +273,73 @@ def _format_date(date):
...
+                self._hyperlink_diff_url(deploy.diff_url),
+            ]
+
+    @property
+    def total_records(self):
+        return HqDeploy.objects.count()
+
+    @cached_property
+    def _user_lookup_url(self):
+        return reverse('web_deploy_lookup')
```
Comment (esoergel) [inline corehq/apps/hqadmin/reports.py:315]: This method doesn't apply to this class - guessing it was copied over from the User List report.

File: corehq/apps/hqadmin/reports.py [hunk 2]
```diff
@@ -266,3 +273,73 @@ def _format_date(date):
...
+
+    def _format_date(self, date):
+        if date:
+            raw_time_since_deploy = dt.now() - date
+            delta_dict = self._strfdelta(raw_time_since_deploy)
```
Comment (esoergel) [inline corehq/apps/hqadmin/reports.py:320]: It looks like this is a reimplementation of the `naturaltime` template filter.  You can actually just call that directly as is done here:

https://github.com/dimagi/commcare-hq/blob/ef2fe2659b3a2d3c0f6b8fcf7274f939321dfadd/corehq/apps/export/views/list.py#L316-L317

File: corehq/apps/hqadmin/reports.py [hunk 3]
```diff
@@ -266,3 +273,73 @@ def _format_date(date):
...
+    def _strfdelta(self, tdelta):
+        # datetime.timedelta object cannot use 'strftime', use custom fn:
+        relative_time_formatted = {"days": tdelta.days}
+        relative_time_formatted["hours"], rem = divmod(tdelta.seconds, 3600)
+        relative_time_formatted["minutes"], relative_time_formatted[
+            "seconds"] = divmod(rem, 60)
+        return relative_time_formatted
+
+    def _hyperlink_diff_url(self, diff_url):
+        return '<a href="{link}">Diff with previous</a>'.format(link=diff_url)
```
Comment (esoergel) [inline corehq/apps/hqadmin/reports.py:345]:
```suggestion
        return f'<a href="{diff_url}">Diff with previous</a>'
```
Happily, HQ is on python 3, so you can use f-strings!

**Review by esoergel — CHANGES_REQUESTED** [review]
Couple minor quibbles, but looks great overall!

**Comment by AliRizvi1** [comment]
Messed up a `git rebase`. Will create a new PR of the same feature/branch.

**Pull request closed without merging**